mne/preprocessing/ica.py

```diff
@@ -884,10 +872,10 @@ def _transform_evoked(self, evoked):
         if not hasattr(self, 'mixing_matrix_'):
             raise RuntimeError('No fit available. Please fit ICA.')
 
-        picks = pick_types(evoked.info, include=self.ch_names, exclude='bads',
-                           meg=False, ref_meg=False)
-
-        if len(picks) != len(self.ch_names):
+        picks = _picks_to_idx(
+            evoked.info, self.ch_names, exclude=[], allow_empty=True)
+        raise_ = False if len(picks) == len(self.ch_names) else True
+        if raise_:
```

Comment:
```suggestion
        if len(picks) != len(self.ch_names):
```

Reply:
Good catch, it was a simplification at the time; I think before I merged the changes to `_picks_to_idx` from main. Completely useless now.